### ISSUE TITLE
Keychain error accessing User data.

### ISSUE THREAD

**Issue by bmctigue** [realm/realm-cocoa#4158]
Hi,

There is some problem accessing any data with the latest build. This is not an issue when I run the sample applications, but they are using old builds.

Basically I can't access any data due to a keychain issue. Here is the code snippet I'm running when first running the app and checking for an existing user:

```swift
func configureDefaultRealm() -> Bool {
    if let user = User.all().first {
        setDefaultRealmConfigurationWithUser(user: user)
        return true
    }
    return false
}
```

The error occurs at accessing the User.all() method.

Here is the error from the log:

> libc++abi.dylib: terminating with uncaught exception of type realm::keychain::KeychainAccessException: Keychain returned unexpected status code: 18446744073709517598

Realm version: ? 2.0.0

Xcode version: ? 8.0

iOS/OSX version: ? 10

Dependency manager + version: ? None, installed frameworks manually

**Comment by bdash** [comment]
The error code `18446744073709517598` looks to be the result of an `OSStatus` error code being misinterpreted as an `uint64_t`. Converting it back to an `OSStatus` reveals the value is `-34018`, or `errSecMissingEntitlement`. I suspect this means your app is missing an entitlement related to the keychain.

**Comment by bdash** [comment]
Discussion at https://forums.developer.apple.com/thread/51071 suggests that the _absence of an entitlements file_ can result in keychain calls failing with `errSecMissingEntitlement` in the simulator. Specifying an entitlements file, even if it contains no entitlements, can work around this problem. @bmctigue, does this workaround work for you?

**Comment by bmctigue** [comment]
Hi, I actually had to turn on KeyChain sharing in the Capabilities section for the app. This created the entitlements file with the setting. I removed the setting to see if just having the file would work, but I received the same error. So turning on Keychain sharing solved this, but I don't think this should be necessary. I'll let you decide this is an issue or not and whether we can close this. Thanks for the help! At least I can move forward...8>)
